docs/contracts/withdrawal-queue-erc721.md

```diff
@@ -3,5 +3,889 @@
+
+:::note
+
+Requirements:
+
+- `stETH` balance of `msg.sender` should be greater than the sum of all `_amounts`
+- permit should have a valid signature, `value` greater than the sum of all `_amounts`, and the `deadline` not expired
+- each amount in `_amounts` should be greater than `MIN_STETH_WITHDRAWAL_AMOUNT` and lower than `MAX_STETH_WITHDRAWAL_AMOUNT`
+
+:::
+
+### requestWithdrawalsWstETHWithPermit()
+
+Batch request the `_amounts` of `wstETH` for withdrawal to the `_owner` address. For each request, the respective amount of `wstETH` is transferred to this contract address, unwrapped to `stETH`, and an `unstETH` NFT is minted to the `_owner` address.`ERC-2612` permit is used to approve the token transfer.
+
+```sol
+function requestWithdrawalsWstETHWithPermit(
+  uint256[] _amounts,
+  address _owner,
+  PermitInput _permit
+) returns (uint256[] requestIds)
+```
+
+where `_permit` is [ERC-2612](https://eips.ethereum.org/EIPS/eip-2612) signed permit structure defined as:
+
+```sol
+    struct PermitInput {
+        uint256 value;
+        uint256 deadline;
+        uint8 v;
+        bytes32 r;
+        bytes32 s;
+    }
+```
+
+Returns the array of ids for each created request. Emits a `WithdrawalRequested` and `Transfer` events.
+
+:::note
+
+Requirements:
+
+- `wstETH` balance of `msg.sender` should be greater than the sum of all `_amounts`
+- permit should have a valid signature, `value` greater than the sum of all `_amounts`, and the `deadline` not expired
+- each amount in `_amounts` should have `getPooledEthByShares(amount)` being greater than  `MIN_STETH_WITHDRAWAL_AMOUNT` and lower than `MAX_STETH_WITHDRAWAL_AMOUNT`
+
+:::
+
+### getWithdrawalRequests()
+
+Returns all withdrawal requests that belong to the `_owner` address
+
+```sol
+function getWithdrawalRequests(address _owner) view returns (uint256[] requestsIds)
+```
+
+:::warning
+
+This operation will copy the entire storage to memory, which can be quite expensive. This method is designed to mostly be used by view accessors that are queried without gas fees. Developers should keep in mind that this function has an unbounded cost, and using it as part of a state-changing function may render the function uncallable if the set grows to a point where copying to memory consumes too much gas to fit in a block.
+
+:::
+
+### getWithdrawalStatus()
+
+Returns `statuses` for requests with ids in `_requestIds`
+
+```sol
+function getWithdrawalStatus(uint256[] _requestIds)
+        view
+        returns (WithdrawalRequestStatus[] statuses)
+```
+
+Returns an array of `WithdrawalRequestStatus` structures, defined as:
+
+```sol
+struct WithdrawalRequestStatus {
+    uint256 amountOfStETH;
+    uint256 amountOfShares;
+    address owner;
+    uint256 timestamp;
+    bool isFinalized;
+    bool isClaimed;
+}
+```
+
+where
+
+- **`amountOfStETH`** — the number of `stETH` tokens transferred to the contract upon request
```

Comment:
not sure why bold in names of struct fields is needed, but not an issue

Reply:
It's for consistency with other docs only.